instrumentation/active_support/lib/opentelemetry/instrumentation/active_support/span_subscriber.rb

```diff
@@ -29,18 +32,36 @@ def self.subscribe(
         )
 
         subscriber_object = ::ActiveSupport::Notifications.subscribe(pattern, subscriber)
+
         ::ActiveSupport::Notifications.notifier.synchronize do
-          if ::Rails::VERSION::MAJOR >= 6
-            s = ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern].pop
-            ::ActiveSupport::Notifications.notifier.instance_variable_get(:@string_subscribers)[pattern].unshift(s)
+          subscribers = find_subscribers(pattern)
+
+          if subscribers.nil?
+            OpenTelemetry.handle_error(
```

Comment:
should this be a debug log rather than an error? a bit of semantics (since the default error handler merely logs the exception.message and message), defer to your judgement.

Reply:
I chose handle_error since it is an unexpected and unrecoverable. 

The intent is to bring it to our attention early on or have users file a bug report.